File: Text/reverse_string/reverse_string.rs
use std::io::stdin;
use itertools::rev;
fn main() {
  println!("Enter the string you want reversed...");
  let mut str = String::new();
  stdin().read_line(&mut str);
  for i in rev(1..str.len()) {
    println!("{}", str.chars().nth(i - 1).unwrap());
  }
}

File: Text/reverse_string/reverse_string.zig
const std = @import("std");
const print = @import("std").debug.print;

pub fn main() !void {
    print("Enter the string you want reversed: ", .{});
    const stdin = std.io.getStdIn().reader();
    var buf: [10]u8 = undefined;
    var input = stdin.readUntilDelimiterOrEof(buf[0..], '\n');
    print("{s}", .{input});
    for (buf) |_, i| {
        const idx = (buf.len - 1) - i;
        print("{c}", .{buf[idx]});
    }
}